### ISSUE TITLE
rustdoc: make linking to examples easier

### ISSUE THREAD

**Issue by alice-i-cecile** [rust-lang/rust#130493]
When writing module-level or other high-level documentation, I often want to link to our examples to show users how to do something in more detail. See https://github.com/bevyengine/bevy/pull/15056 for an example of this problem.

This is distinct from the automatically generated links to examples (which are great!): we want to be able to manually link to specific examples.

@GuillaumeGomez pointed out that this can be done in part using relative paths, e.g. `../../src/custom_loop/custom_loop.rs.html#19`. However, this is very fragile: any change to either file breaks the link with no tooling to detect the breakage. This functionality is also undocumented, and could be broken by rustdoc / docs.rs at any time.

To add full support, he thinks we might need an intra-doc link extension to make it work.

**Comment by alice-i-cecile** [comment]
Yep, that looks great, assuming the dead intradoc links lint triggers correctly.

**Comment by GuillaumeGomez** [comment]
Coming back to this, this disambiguator syntax `@` would work too and would be coherent with the rest:

```rust
/// Oh what a nice [example][example@example_name]!
```

**Comment by GuillaumeGomez** [comment]
Opened the PR to add this feature: https://github.com/rust-lang/rust/pull/132792
